mlx/ops.h

```diff
@@ -400,6 +400,14 @@ array where(
     const array& y,
     StreamOrDevice s = {});
 
+/** Replace NaN and infinities with finite numbers. */
+array nan_to_num(
+    const array& a,
+    const float& nan,
+    const float& posinf,
+    const float& neginf,
```

Comment:
I think these should be `std::optional<float>` so that we can have the default behavior in the C++ API.